api/envoy/type/matcher/v3/regex.proto

```diff
@@ -48,11 +49,22 @@ message RegexMatcher {
         [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
   }
 
+  message RegexEngine {
```

Comment:
Instead of adding this custom message type, I think it would make sense to instead use the existing `xds.core.v3.TypedExtensionConfig` type:

https://github.com/cncf/xds/blob/d92e9ce0af512a73a3a126b32fa4920bee12e180/xds/core/v3/extension.proto#L14

That having been said, I am not sure that it really makes sense to add a new extension point specifically for the regex engine in the first place.  Instead, why not simply define a new matcher for the new regex engine?  That way, you can plug into the existing extension point in the new matcher API:

https://github.com/cncf/xds/blob/d92e9ce0af512a73a3a126b32fa4920bee12e180/xds/type/matcher/v3/matcher.proto#L63

Reply:
I think it is a good idea, and I will take a look at the custom match. Thanks.